workshop-todolist/index.js

```diff
@@ -0,0 +1,217 @@
+    const tasks = getTasksFromLocalStorage();
+
+    // Проходимо по масиву задач і додаємо кожну задачу в список, в DOM
+    tasks.forEach((task) => {
+        createSingleTaskElement(task.text, task.id);
+    })
+}
+
+/**
+ * Створюємо окрему задачу
+ * @param {Event} event - The triggering event
+ */
+function createTask(event) {
+    // Блокуємо дефолтний сабміт форми
+    event.preventDefault();
+    // Виходимо з функції якщо в полі немає тексту і видаляймо непотрібні пробіли до і після тексту
+    if (taskInput.value.trim() === '') {
+        return;
+    }
+    const taskId = generateUUID();
+    // Створюємо нову задачу і додаємо в DOM
+    createSingleTaskElement(taskInput.value, taskId);
+    // Додаємо нову задачу в localStorage
+    storeTaskInLocalStorage(taskInput.value, taskId);
+    // Очищуємо поле після додавання нової задачі в список
+    taskInput.value = '';
+}
+
+/**
+ * Додаємо нову створену задачу в localStorage
+ * @param {String} task - окрема задача
+ * @param {String} taskId - унікальний ідентифікатор задачі
+ */
+function storeTaskInLocalStorage(task, taskId) {
+    // Отримуємо поточні задачі з localStorage
+    const tasks = getTasksFromLocalStorage();
+    const taskObject = {
+        id: taskId,
+        text: task,
+    };
+    // Додаємо нову задачу в масив
+    tasks.push(taskObject);
+    // Записуємо оновлений масив в localStorage
+    setTasksToLocalStorage(tasks);
+}
+
+/**
+ * Видаляємо всі задачі з localStorage та з DOM
+ */
+function clearAllTasks() {
+    // Показуємо користувачу модальне вікно для підтвердження видалення всіх задач
+    if (confirm('Ви впевнені що хочете видалити всі задачі?')) {
+        // Якщо користувач підтверджує, то видаємо всі задачі з localStorage та з DOM
+        localStorage.clear();
+        taskList.innerHTML = '';
+    }
+}
+
+/**
+ * Видаляємо окрему задачу з localStorage та з DOM
+ * @param {Event} event - The triggering event
+ 
+function clearSingleTask(event) {
+    // Отримуємо батьківський елемент елементу на якому була подія кліку
+    const iconContainer = event.target.parentElement;
+
+    // Якщо батьківський елемент має відповідний клас
+    if (iconContainer.classList.contains('delete-item')) {
+        // Отримуємо підтвердження користувача
+        if (confirm('Ви впевнені що хочете видалити цю задачу?')) {
+            // Видаляємо елемент з DOM та з localStorage
+            const taskItem = iconContainer.parentElement;
+            const taskId = taskItem.dataset.taskId;
+            taskItem.remove();
+            removeTaskFromLocalStorage(taskId);
+        }
+    }
+}
+*/
+
+/**
+ * Обробник подій для видалення та редагування задачі
+ * @param {Event} event - The triggering event
+ */
+function handleTaskAction(event) {
+    const target = event.target;
+    const taskItem = target.parentElement;
+    const taskId = taskItem.dataset.taskId;
+  
+    if (target.classList.contains('delete-item')) {
```

Comment:
З того що тут зараз написано, почніть з того що перевірка класу не на тому компоненті взагалі, має бути так
```suggestion
    if (taskItem.classList.contains('delete-item')) {
```
Відштовхуючись від цього спробуйте далі розібратись, що у вас відбувається в подальшій логіці